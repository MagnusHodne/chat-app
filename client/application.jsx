import { BrowserRouter, Route, Routes } from "react-router-dom";
import React, { useContext } from "react";
import { FrontPage } from "./pages/frontPage";
import {
  ErrorComponent,
  LoadingComponent,
} from "./components/feedbackComponents";
import { useLoading } from "./lib/useLoading";
import { LoginPage } from "./pages/loginPage";
import { LoginForm } from "./components/loginForm";
import { ChatApiContext } from "./chatApiContext";

export function Application() {
  const { fetchLogin } = useContext(ChatApiContext);
  //Storing user as a global variable
  const { data, error, loading, reload } = useLoading(fetchLogin);

  if (loading) {
    return <LoadingComponent message={"Fetching user data, please wait..."} />;
  }
  if (error) {
    return <ErrorComponent error={error} />;
  }

  return (
    <BrowserRouter>
      <Routes>
        <Route path={"/startlogin"} element={<LoginForm />} />
        <Route path={"/*"} element={<FrontPage user={data?.user} />} />
        <Route
          path={"/login/*"}
          element={<LoginPage config={data?.config} reload={reload} />}
        />
      </Routes>
    </BrowserRouter>
  );
}
