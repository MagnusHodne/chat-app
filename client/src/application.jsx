import { BrowserRouter, Route, Routes } from "react-router-dom";
import React, { useContext, useEffect, useState } from "react";
import { FrontPage } from "./pages/frontPage";
import {
  ErrorComponent,
  LoadingComponent,
} from "./components/feedbackComponents";
import { useLoading } from "./lib/useLoading";
import { LoginPage } from "./pages/loginPage";
import { LoginForm } from "./components/loginForm";
import { ChatApiContext } from "./chatApiContext";
import { UserContext } from "./userContext";
import { useAuth0 } from "@auth0/auth0-react";

export function Application() {
  const { isLoading, user } = useAuth0();
  const [userInfo, setUserInfo] = useState(null);

  useEffect(() => {
    if (!isLoading && user) {
      setUserInfo(user);
    }
  }, [isLoading, user]);
  //Storing user as a global variable

  if (isLoading) {
    return <LoadingComponent message={"Fetching user data, please wait..."} />;
  }

  return (
    <UserContext.Provider value={{ user }}>
      <BrowserRouter>
        <Routes>
          <Route path={"/"} element={<LoginForm />} />
          <Route path={"/*"} element={<FrontPage />} />
        </Routes>
      </BrowserRouter>
    </UserContext.Provider>
  );
}
