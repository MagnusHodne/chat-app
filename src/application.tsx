import { Route, Routes } from "react-router-dom";
import { FrontPage } from "./pages/frontPage";
import {
  ErrorComponent,
  LoadingComponent,
} from "./components/feedbackComponents";
import { LoginForm } from "./components/loginForm";
import { useAuth0 } from "@auth0/auth0-react";
import { LoginCallback } from "./pages/loginCallback";

export function Application() {
  const { isLoading, error, isAuthenticated } = useAuth0();
  if (isLoading) {
    return <LoadingComponent message={"Authenticating user, please wait..."} />;
  }

  if (error) {
    return (
      <ErrorComponent error={`Error during authentication: ${error.message}`} />
    );
  }

  if (!isAuthenticated) {
    return <LoginForm />;
  }
  return (
    <Routes>
      <Route path={"/*"} element={<FrontPage />} />
      <Route path={"/oauth2/callback"} element={<LoginCallback />} />
    </Routes>
  );
}
