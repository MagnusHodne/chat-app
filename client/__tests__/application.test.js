import React from "react";
import ReactDOM from "react-dom";
import { Application } from "../application";

describe("Application", () => {
  it("renders correctly", () => {
    const element = document.createElement("div");
    ReactDOM.render(<Application />, element);
    expect(element).toMatchSnapshot();
  });
});

//See reference 10 for async test setup
