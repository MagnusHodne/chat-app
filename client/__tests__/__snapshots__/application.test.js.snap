// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application renders correctly 1`] = `
<div>
  <div
    class="loading-component"
  >
    <div>
      <span
        class="fa-solid fa-spinner fa-spin-pulse"
      />
      <p>
        Fetching user data, please wait...
      </p>
    </div>
  </div>
</div>
`;
